cypress/integration/Switch.spec.ts

```diff
@@ -0,0 +1,58 @@
+import * as h from '../helpers';
+
+const getSwitch = () => {
+  return cy.get(`[type="checkbox"]`);
+};
+
+describe('Non-disabled Switch', () => {
+  before(() => {
+    h.stories.visit();
+  });
+  ['Default', 'Alert', 'Error'].forEach(story => {
+    context(`Given the '${story}' story is rendered`, () => {
+      beforeEach(() => {
+        h.stories.load('Components|Inputs/Switch/React/Top Label', story);
+      });
+
+      it('should not have any axe errors', () => {
+        cy.checkA11y();
+      });
+
+      context('when the tab key is pressed', () => {
+        beforeEach(() => {
+          cy.tab();
+        });
+        it('should be the focused item on the page', () => {
+          getSwitch().should('have.focus');
+        });
+      });
+
+      context('when the switch is clicked', () => {
+        beforeEach(() => {
+          getSwitch().click();
+        });
+        it('should be checked', () => {
+          getSwitch().should('be.checked');
```

Comment:
Cypress tests are about specifications from a user's perspective. This spec reads in full "Non-disabled Switch when the switch is clicked should be checked". Linking the given of the story being rendered, the when of the switch being clicked, and the then of the side effect of that action where the switch is now checked.

I might argue that "checked" is an implementation detail of a switch being "on", but there is a bit of controversy around how a "switch" should be labelled/read by screen readers from an accessibility standpoint. I don't remember the final say the `role` attribute of a `Switch` which effects how it will be read by a screen reader.

I personally avoid actions in unit-level tests including clicks. Most of the time firing a `click` event is enough for the semantic action we'd call "click", but some components will expect other browser events in a specific order and some events aren't as simple as a "click". I leave that all for Cypress which has wiring for all that. In Cypress, a "click" includes other events normally associated with "click" like "mouseover". I've seen unit tests that try to recreate all that. Yuck.

I wouldn't worry about possible duplication in testing levels. Each serve a specific purpose.